pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -70,6 +75,11 @@ func NewNamespaceController(kubeClient kube.Client, caBundleWatcher *keycertbund
 		controllers.WithReconciler(c.reconcileCACert),
 		controllers.WithMaxAttempts(maxRetries))
 
+	c.ignoredNamespaces = inject.IgnoredNamespaces.Copy()
+	if platform.IsOpenShift() && features.EnableAmbientControllers {
```

Comment:
I'm not sure only OpenShift need this, but can we make this configurable?

Reply:
I think on GKE you need to install in kube-system as well so this seems like it applies there as well

Reply:
> can we make this configurable?

Actually, I wanted to avoid adding another environment variable to not complicate it.

> I think on GKE you need to install in kube-system as well so this seems like it applies there as well

Do you suggest to include GKE in this condition or to make it configurable?

Reply:
TBH globally-ignored namespace defaults shouldn't be configured in code at all.

There is no actual "default", it's very contingent and it could easily be a helm default value.

Reply:
I agree with make it configurable. Donot introduces platform concept any more. In agent @howardjohn  has removed vendor plugin

Reply:
Okay, thanks for sharing your opinions. I wonder if it would be better to make `inject.IgnoredNamespaces` configurable or to introduce another env variable, like `ROOT_CA_SKIP_NAMESPACES`, which would be used only by the namespace controller. I'm thinking about the second option, because we need it only for ztunnel and I don't know if I should change injection rules. Can you share your thoughts on that?

Reply:
I'm inclined to actually make `IgnoredNamespaces` use config-based defaults rather than code-based ones. 

That's how it should work, and adding another new env var scoped just to this problem just because injection defaults are hardcoded (and should not be/do not need to be) seems wrong to me. 

If you did add `ROOT_CA_SKIP_NAMESPACES` it would be a strict superset of `inject.IgnoredNamespaces` in all cases, right?

Would there be *any* scenario where a namespace should be denylisted in `inject.IgnoredNamespaces` but _not_ in `ROOT_CA_SKIP_NAMESPACES`, or vice versa? I can't think of one.

Reply:
> Would there be any scenario where a namespace must be denylisted in inject.IgnoredNamespaces but must not be in ROOT_CA_SKIP_NAMESPACES, or vice versa? I can't think of one.

Unless I am getting things mixed up, I think the use case is ztunnel in kube-system but no sidecars there?

Reply:
> Unless I am getting things mixed up, I think the use case is ztunnel in kube-system but no sidecars there?

Yes, exactly.

> That's how it should work, and adding another new env var scoped just to this problem just because injection defaults are hardcoded (and should not be/do not need to be) seems wrong to me.

Ok, thanks. I was thinking about that, because the goal is making ztunnel work in kube-system, not changing current injection semantics. And if `inject.IgnoredNamespaces` is still hard-coded, I guess users don't need it. That's why I was considering `ROOT_CA_SKIP_NAMESPACES`.

Reply:
> Unless I am getting things mixed up, I think the use case is ztunnel in kube-system but no sidecars there?

If we ever want to deploy ztunnel to a namespace where we **_would_** also explicitly want to inject sidecars, I can see an argument for separate config keys. Otherwise, I think making `IgnoredNamespaces` customizable is better.

Another option: simply remove namespace constraints entirely for ztunnel, for all platforms. It should be trivial to disable injection for ztunnel pods (we might already do that) - and I can't think of any other reason why we would want to force specific namespaces for it.

Reply:
maybe we should use annotation/label instead of environment variable?

Reply:
Annotation/label where ? On namespace ? That would be pretty good - env variables ( and mesh config !) were meant for experimental things and feature flags, not as API. My concern is that I still don't understand if this is for some multi-tenant support or how user should use it if it becomes an API.